tests/integration/records/directory_validator.py

```diff
@@ -48,25 +52,33 @@ def validate_records_schema(self):
                                     'timestamp', 'timestamptz']
             assert actual_field_names == expected_field_names, actual_field_names
             actual_field_types = [field['type'] for field in records_schema['fields'].values()]
-            acceptable_field_types = [
-                # gold star if you can do all of this
-                [
-                    'integer', 'string', 'string', 'string',
-                    'string', 'string', 'string', 'date', 'time',
-                    'datetime', 'datetimetz'
-                ],
-                # Redshift doesn't support TIME type:
-                # https://docs.aws.amazon.com/redshift/latest/dg/r_Datetime_types.html
-                [
-                    'integer', 'string', 'string', 'string',
-                    'string', 'string', 'string', 'date', 'string',
-                    'datetime', 'datetimetz'
-                ]
+            field_types_are_ok = False
+            # gold star if you can do all of this
+            ideal_field_types = [
+                'integer', 'string', 'string', 'string',
+                'string', 'string', 'string', 'date', 'time',
+                'datetime', 'datetimetz'
             ]
-            assert actual_field_types in acceptable_field_types,\
-                f"\nreceived {actual_field_types}, \nexpected {acceptable_field_types}"
+            if actual_field_types == ideal_field_types:
+                field_types_are_ok = True
+            else:
+                acceptable_field_types_by_db = {
+                    # Redshift doesn't support TIME type:
+                    # https://docs.aws.amazon.com/redshift/latest/dg/r_Datetime_types.html
+                    'redshift': [
+                        'integer', 'string', 'string', 'string',
+                        'string', 'string', 'string', 'date', 'string',
+                        'datetime', 'datetimetz'
+                    ]
+                }
+                if actual_field_types == acceptable_field_types_by_db[self.source_db_type]:
```

Comment:
Not a big deal, this will raise an exception if, say, there is a regression in vertica field handling. Obviously that yields the same result as raising an assertion, but with a different message.

Something like this would let it make it to the assertion:

```suggestion
                if actual_field_types == acceptable_field_types_by_db.get(self.source_db_type):
```

Reply:
Sounds good!